File: cpp/165_CompareVersionNumbers.cpp
class Solution {
public:
    int compareVersion(string version1, string version2) {
        std::vector<string> sv1;
        std::vector<string> sv2;
        split(version1, '.',sv1);
        split(version2, '.',sv2);
        int i = 0;
        for(; i < sv1.size(); ++i){
            if(i >= sv2.size()){
                 if(std::stoi(sv1[i]) != 0) return 1;
                 continue;
            }
            int v1 = std::stoi(sv1[i]);
            int v2 = std::stoi(sv2[i]);
            if(v1 > v2) return 1;
            if(v1 < v2) return -1;
        }
        while(i < sv2.size()){
            if(std::stoi(sv2[i]) != 0) return -1;
            i++;
        }
        return 0;

    }
private:
     void split(const std::string &s, char delim, std::vector<string>& stringList){
        std::stringstream ss(s);
        std::string item;
        while(std::getline(ss, item, delim)){
            stringList.push_back(item);
        }
    }
};

File: js/165_CompareVersionNumbers.js
/**
 * @param {string} version1
 * @param {string} version2
 * @return {number}
 */
var compareVersion = function(version1, version2) {
    var vl1 = version1.split('.');
    var vl2 = version2.split('.');
    console.log("v1:"+vl1+" v2:"+vl2);
    var i = 0;
    while( i < vl1.length ){
        if(i >= vl2.length){
            if(parseInt(vl1[i]) !== 0) return 1;
        }else{
            if(parseInt(vl1[i]) > parseInt(vl2[i])) return 1;
            else if(parseInt(vl1[i]) < parseInt(vl2[i])) return -1;
        }
        ++i;
    }
    while( i < vl2.length ){
        if(parseInt(vl2[i]) !== 0 ) return -1;
        ++i;
    }
    return 0;
};